src/mock/MockEIP712WithCustomChainID.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+/// @notice EIP712 helpers for permit2
+/// @dev Maintains cross-chain replay protection in the event of a fork
+/// @dev Reference: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/EIP712.sol
+contract MockEIP712WithCustomChainID {
```

Comment:
Why did you copy-paste the code from EIP712 and not just inherit EIP712 contract? I don't think that there should be any copied line of code if you can inherit based tested contract. Imagine that these tests will be used for a while and then Permit2 will be updated, so tests won't cover the new logic of the contract, but only the old one. The amount of code for reviewer is also bigger, as well as supporting it is very hard

```suggestion
import "../EIP712.sol";

/// @notice EIP712 helpers for permit2
/// @dev Maintains cross-chain replay protection in the event of a fork
/// @dev Reference: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/EIP712.sol
contract MockEIP712WithCustomChainID is EIP712 {
```

Reply:
In this case, you not only need to inherit but also override the function `_buildDomainSeparator`

Reply:
Thanks for your comment. I will try to create a contract inherited from EIP712, override _buildDomainSeparator() and use it in tests. This should improve security and simplify test maintenance.

Reply:
Since the function is not "virtual" it is not possible, so just leave it as it is